### PR TITLE
fix(Core/Creature): Remove automated z spawn correction

### PR DESCRIPTION
**Pull request by sudlud**
## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

follow-up to #18153
functionality removed as suggested by @Nyeriah 

If spawns are stuck in the ground imo these should be fixed in DB and not rely on this automated manipulation.

I've assembled a list of creature spawns that used to trigger this function as of 4cfad0ffa:
https://gist.github.com/sudlud/b57f3ee6617e2cf10fe2edc554c13245

I've randomly checked some of these spawns - at least for creatures with movement it doesn't seem to matter, they still wander around.

Also there's at least one (sniffed) spawn that could never spawn correctly while this functionality is implemented as it's beeing spawned in a room directly under the ceiling:

[DND] Dalaran Toy Store Plane String Hook
`(106190, 29807, 571, 1, 1, 0, 5812.986328125, 686.76739501953125, 655.3192138671875, 0, 120, 0, 0, 0, 0, 0, "", 48632, 1, NULL);
`

(The automated correction then moved it out of the room above the ceiling.)

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by sudlud** [comment]
looks like no creature spawn of map 0 and 1 was affected 👀

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -1627,7 +1627,7 @@ bool Creature::isVendorWithIconSpeak() const
     return m_creatureInfo->IconName == "Speak" && m_creatureData->npcflag & UNIT_NPC_FLAG_VENDOR;
 }
 
-bool Creature::LoadCreatureFromDB(ObjectGuid::LowType spawnId, Map* map, bool addToMap, bool gridLoad, bool allowDuplicate /*= false*/)
+bool Creature::LoadCreatureFromDB(ObjectGuid::LowType spawnId, Map* map, bool addToMap, bool /*gridLoad*/, bool allowDuplicate /*= false*/)
```
Comment (Kitzunu) [inline src/server/game/Entities/Creature/Creature.cpp:1630]: why keep gridLoad if we wont use it

**Comment by Kitzunu** [comment]
only 10k spawns, that's manageable kek

File: src/server/game/Entities/Creature/Creature.cpp (hunk 1 above)
Comment (sudlud) [reply src/server/game/Entities/Creature/Creature.cpp:1630]: True, will remove it later today

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**